crates/oxc_linter/src/rules/unicorn/prefer_optional_catch_binding.rs

```diff
@@ -0,0 +1,119 @@
+use oxc_ast::{
+    ast::{BindingPattern, BindingPatternKind},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(prefer-optional-catch-binding): Prefer omitting the catch binding parameter if it is unused")]
+#[diagnostic(severity(warning))]
+struct PreferOptionalCatchBindingDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferOptionalCatchBinding;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Prefers omitting the catch binding parameter if it is unused
+    ///
+    /// ### Why is this bad?
+    ///
+    /// It is unnecessary to bind the error to a variable if it is not used.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// try {
+    ///  // ...
+    /// } catch (e) { }
+    ///
+    /// // Good
+    /// try {
+    ///  // ...
+    /// } catch { }
+    /// ```
+    PreferOptionalCatchBinding,
+    style
+);
+
+impl Rule for PreferOptionalCatchBinding {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::CatchClause(catch_clause) = node.kind() else { return };
+
+        let Some(catch_param) = &catch_clause.param else { return };
+
+        let references_count = get_param_references_count(catch_param, ctx);
+
+        if references_count != 0 {
+            return;
+        }
+
+        ctx.diagnostic(PreferOptionalCatchBindingDiagnostic(catch_param.span()));
+    }
+}
+
+fn get_param_references_count(binding_pat: &BindingPattern, ctx: &LintContext) -> usize {
+    match &binding_pat.kind {
```

Comment:
There's a `BoundNames` trait that can be used to get all the binding identifiers of a patter https://github.com/web-infra-dev/oxc/blob/main/crates/oxc_ast/src/syntax_directed_operations/bound_names.rs

Reply:
nice

Reply:
I guess, since BoundNames doesn't traverse the assignment part of the AssignmentPattern, we can't know what's there.

If is something with, or possibly with side effects, we probably shouldn't report